controllers/csi-migrator/persistentvolume_controller.go

```diff
@@ -98,8 +98,22 @@ func (r *PersistentVolumeReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, nil
 	}
 
+	_, isSourceReplicated := storageClass.Parameters[controller.StorageClassReplicationParam]
+	_, isTargetReplicated := targetStorageClass.Parameters[controller.StorageClassReplicationParam]
+	var migrateType migration.MigrateTypes
+	switch true {
+	case (isSourceReplicated && isTargetReplicated) || (!isSourceReplicated && !isTargetReplicated):
+		migrateType = migration.MigrateTypes_VERSION_UPGRADE
+	case !isSourceReplicated && isTargetReplicated:
+		migrateType = migration.MigrateTypes_NON_REPL_TO_REPL
+	case isSourceReplicated && !isTargetReplicated:
+		migrateType = migration.MigrateTypes_REPL_TO_NON_REPL
+	default:
+		log.Info("Strange migration type..")
```

Comment:
Don't you think we should do more than that? Perhaps just return non nil, removing annotation and saying that we can't figure out migration type?